src/iceberg/name_mapping.cc

```diff
@@ -0,0 +1,267 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/name_mapping.h"
+
+#include <format>
+#include <sstream>
+
+#include "iceberg/util/formatter_internal.h"
+
+namespace iceberg {
+
+namespace {
+
+// Helper function to join a list of field names with a dot
+std::string JoinByDot(std::span<const std::string> parts) {
```

Comment:
I guess you can use [`join_view`](https://en.cppreference.com/w/cpp/ranges/join_view) for this too

Reply:
I tried it but it seems that Clang does not yet implement this.